corehq/apps/api/resources/v0_1.py

```diff
@@ -156,15 +158,43 @@ class Meta(UserResource.Meta):
         object_class = WebUser
         resource_name = 'web-user'
 
+    def _property_filter(self, field_name, gt=None, gte=None, lt=None, lte=None):
+        def filter(docs):
+            return [
+                doc for doc in docs
+                if field_name in doc and doc[field_name] is not None
+                and all([
+                    (gt is None or doc[field_name] > gt),
+                    (gte is None or doc[field_name] >= gte),
+                    (lt is None or doc[field_name] < lt),
+                    (lte is None or doc[field_name] <= lte)
+                ])
+            ]
+        return filter
+
     def obj_get_list(self, bundle, **kwargs):
         domain = kwargs['domain']
         username = bundle.request.GET.get('web_username')
+
+        COMPOUND_FILTERS = {
+            'last_modified': make_date_filter(functools.partial(self._property_filter, 'last_modified'))
+        }
+        filters = []
+        for key in bundle.request.GET.keys():
+            if '.' in key and key.split(".")[0] in COMPOUND_FILTERS:
+                prefix, qualifier = key.split(".", maxsplit=1)
+                filters.append(COMPOUND_FILTERS[prefix](qualifier, bundle.request.GET.get(key)))
+
         if username:
             user = WebUser.get_by_username(username)
             if not (user and user.is_member_of(domain) and user.is_active):
                 user = None
-            return [user] if user else []
-        return list(WebUser.by_domain(domain))
+            results = [user] if user else []
+        else:
+            results = list(WebUser.by_domain(domain))
+        for filter in filters:
+            results = filter(results)
+        return results
```

Comment:
I don't think filtering in memory will be a sustainable model if this is going to see heavy use, as we still have to fetch all users on the domain to do it this way.  I think it's worth modifying this to use the `UserQuerySetAdapter` and pulling from ES, like `CommCareUserResource` does.  That'll also give us `prime_user_data_caches` for free, which is another ticket we have in the mix somewhere.  That's of course a bigger task, so we might want to create a new ticket for it.

Reply:
I had an iffy feeling about doing it in memory too. As is, this change would address the immediate issue - where there is too much data for the lamba function to handle. But agreed that it isn't a sustainable strategy.

> That'll also give us prime_user_data_caches for free, which is another ticket we have in the mix somewhere
I believe this is the ticket you're referring to [USH-5529](https://dimagi.atlassian.net/browse/USH-5529). Since i'll be implementing the `UserQuerySetAdapter` there, how about i also include the refactor to filter using ES. I think that'll mainly be creating a new filter like the ones [here](https://github.com/dimagi/commcare-hq/blob/e2580a67c4ee94ce299016a1b20b50025a8f6186/corehq/apps/es/users.py#L44-L62)

Reply:
In this PR, I included the ES migration needed to add `last_modified` to user mapping. In this followup [PR](https://github.com/dimagi/commcare-hq/pull/35868), I created the filter and used ES to filter `last_modified`, instead of in memory.